### DISCUSSION TITLE
Using Create New Note From Template in the File Explorer created the note in the left pane

### DISCUSSION THREAD

**Discussion by Randy-Astle** [SilentVoid13/Templater#783] [Help]
**Plugin information (please complete the following information):**

-   OS: Windows 10
-   Templater version: 1.12.0
-   Obsidian version: 0.15.2
-   Templater settings: 
    - Template folder location: "Templates/",
    - Syntax highlighting: enabled
    - Automtic jump to cursor: disabled
    - Trigger Templater on New File Creation: disabled 

**Describe the bug**
I have template to create a note to store account information.

When I used it via the Create New Note From Template context menu in the File Explorer, the note was created in the left panel.

**Expected behavior**
Expected the new note to be placed in the active editor pane.

**Screenshots**

**Additional context**

**Comment by welpdx** [comment]
Please see #670. 
The code snippet below seemed to work if added at the end of your template:
```js
<%*
if (this.app.workspace.activeLeaf.getViewState()?.type =="file-explorer") {
this.app.workspace.setActiveLeaf(this.app.workspace.getMostRecentLeaf())
}
%>
```
<details>
           <summary style="font-size:20">Demo and explanation</summary>
           <img src="https://i.imgur.com/jJxLeIR.gif" alt="demo" />
<p>If the current active leaf == the left file explorer pane, switch the focus to the most recent leaf. </p>
</details>

Hope that helps.

~welp

**Comment by diale13** [comment]
I have the same problem!

The script doesent work for me and it keeps opening on top of the File explorer

**Comment by welpdx** [comment]
@diale13  can you tell me exactly what steps you took that ended up with this problem? Is it the same steps as the op? (i.e., focus is on the file explorer, and then Templater: create new note is executed)
And what os are you using? 


Also, what is the template you are using? Can you try adding the above script at the start of your template and see what happens?
Thanks

**Comment by diale13** [comment]
Yes sure, sorry i took a while to respond.

OS: Windows 10
Templater version: 1.12.0
Obsidian version: 0.15.8

Templater settings:
        Template folder location: "📜Templates/",
        Syntax highlighting: disabled
        Automtic jump to cursor: disabled
        Trigger Templater on New File Creation: disabled

**Describe the bug**

Process:
![imagen](https://user-images.githubusercontent.com/33520681/180660057-8f05ca42-d98f-46eb-aa2d-f3683b32557e.png)

After template gets created home bar gets replaced by the template. Only way to restore home side bar is to close Obsidian.
![imagen](https://user-images.githubusercontent.com/33520681/180660067-762109c3-7c9a-4e7f-899a-9caf9fde3dd3.png)

When I used it via the Create New Note From Template context menu in the File Explorer, the note was created in the left panel.

**Expected behavior**
Expected the new note to be placed in the active editor pane.


Thanks

**Comment by AB1908** [comment]
Is #691 related?

**Comment by welpdx** [comment]
@diale13  until this issue is resolved, please just add this script to the end of your template
```js
<%*
if (this.app.workspace.activeLeaf.getViewState()?.type =="file-explorer") {
this.app.workspace.setActiveLeaf(this.app.workspace.getMostRecentLeaf())
}
%>
``` 
![img](https://i.imgur.com/Em8b8mU.gif)

The alternative to completely closing obsidian is to just `Reload` from command palate or F5 from console.

**Comment by AB1908** [comment]
This should be fixed by #736.
